app-creator/src/widgets/dropzone-widget/dropzone-widget.ts

```diff
@@ -0,0 +1,226 @@
+      display: flex;
+      flex-direction: column;
+      align-items: center;
+      justify-content: center;
+      overflow-y: scroll;
+    }
+
+    p {
+      margin: 0px;
+      margin-top: var(--tight);
+      font-size: 0.8rem;
+    }
+
+    #empty-placeholder-icon {
+      color: var(--border-gray);
+    }
+
+    #empty-placeholder-text {
+      color: var(--border-gray);
+    }
+
+    #empty-placeholder {
+      display: flex;
+      align-items: center;
+      justify-content: center;
+      flex-direction: column;
+    }
+  `;
+
+  /**
+   * Additional custom styles for the button.
+   */
+  @property({ type: Object }) styles = {};
+
+  /**
+   * Callback triggered on input change.
+   */
+  @property({ type: Object })
+  debouncedDragOver: (e: Event) => void = () => {};
+
+  render() {
+    const { styles, handleDragOver, handleDragenter, handleDrageleave } = this;
+
+    const emptyPlaceholder = html`
+      <div id="empty-placeholder">
+        <iron-icon
+          id="empty-placeholder-icon"
+          icon="icons:system-update-alt"
+        ></iron-icon>
+        <p id="empty-placeholder-text">Drop widgets here</p>
+      </div>
+    `;
+
+    return html`
+      <div
+        id="container"
+        @dragenter=${handleDragenter}
+        @dragleave=${handleDrageleave}
+        @dragover=${handleDragOver}
+        style="${styleMap(styles)}"
+      >
+        ${emptyPlaceholder}
+      </div>
+    `;
+  }
+
+  /**
+   * Callback triggered whenever we drag a widget over a dropzone-widget.
+   */
+  handleDragOver(e: DragEvent) {
+    // get container element
+    let container = this.shadowRoot?.getElementById('container');
+    if (container == null) {
+      return;
+    }
+
+    // get widget that's currently being dragged
+    const widget = store.draggingElement as HTMLElement;
+    const widgetWrapper = widget.parentElement;
+    if (widget == null || widgetWrapper == null) {
+      return;
+    }
+
+    // get next element
+    const nextElement = this.getNextElement(widget, e.clientY);
+
+    // in case we are reordering an element, we want to move the actual element rather than creating a clone
+    const reordering = (widgetWrapper as DraggableWidget).editable;
+    if (reordering) {
+      if (nextElement == null) {
+        container.appendChild(widgetWrapper);
+      } else {
+        container.insertBefore(widgetWrapper, nextElement);
+      }
+      return;
+    }
+
+    // making clone with new id
+    const clone = widget.cloneNode(true) as HTMLElement;
+    clone.id += `-${store.widgetIDs[widget.id]}`;
+
+    // check if the element already exists
+    // this is necessary because the event is fired multiple times consecutively
+    const element = this.shadowRoot?.getElementById(`${clone.id}-wrapper`);
+    if (element != null) {
+      container.removeChild(element);
+    }
+
+    // the cloned widget is not wrapped with a draggable widget so we have to create one below
```

Comment:
Could we extract this into a new function and call it something like wrapDraggableWidget?